PWGDQ/Tasks/dqEfficiency.cxx

```diff
@@ -835,13 +836,210 @@ struct AnalysisSameEventPairing {
+      //fOutputList.setObject(fHistMan->GetMainHistogramList());
+
+      histNames += "DileptonsSelected;DileptonTrackInvMass;";
+
+      if (!sigNamesStr.IsNull()) {
+        for (int isig = 0; isig < objRecSigArray->GetEntries(); ++isig) {
+          MCSignal* sig = o2::aod::dqmcsignals::GetMCSignal(objRecSigArray->At(isig)->GetName());
+          if (sig) {
+            if (sig->GetNProngs() == 1) {
+              fRecMCSignals.push_back(*sig);
+              TString histName = Form("LeptonsSelected_matchedMC_%s", sig->GetName());
+              histNames += Form("%s;", histName.Data());
+              fRecMCSignalsNames.push_back(sig->GetName());
+            }
+            if (sig->GetNProngs() == 2) {
+              fRecMCSignals.push_back(*sig);
+              TString histName = Form("DileptonsSelected_matchedMC_%s", sig->GetName());
+              histNames += Form("%s;", histName.Data());
+              fRecMCSignalsNames.push_back(sig->GetName());
+            }
+            if (sig->GetNProngs() == 3) {
+              fRecMCSignals.push_back(*sig);
+              TString histName = Form("DileptonTrackInvMass_matchedMC_%s", sig->GetName());
+              histNames += Form("%s;", histName.Data());
+              fRecMCSignalsNames.push_back(sig->GetName());
+            }
+          }
+        }
+      }
+
+      DefineHistograms(fHistMan, histNames.Data()); // define all histograms
+      VarManager::SetUseVars(fHistMan->GetUsedVars());
+      fOutputList.setObject(fHistMan->GetMainHistogramList());
+    }
+
+    TString configCutNamesStr = fConfigTrackCuts.value;
+    if (!configCutNamesStr.IsNull()) {
+      std::unique_ptr<TObjArray> objArray(configCutNamesStr.Tokenize(","));
+      fNHadronCutBit = objArray->GetEntries();
+    } else {
+      fNHadronCutBit = 0;
+    }
+  }
+
+  // Template function to run pair - track combinations
+  template <int TPairType, uint32_t TEventFillMap, uint32_t TEventMCFillMap, uint32_t TTrackFillMap, typename TEvent, typename TTracks, typename TEventsMC, typename TTracksMC>
+  void runDileptonLepton(TEvent const& event, TTracks const& tracks, soa::Join<aod::Dileptons, aod::DileptonsExtra> const& dileptons, TEventsMC const& eventsMC, TTracksMC const& tracksMC)
+  {
+    VarManager::ResetValues(0, VarManager::kNVars, fValuesTrack);
+    VarManager::ResetValues(0, VarManager::kNVars, fValuesDilepton);
+    VarManager::FillEvent<TEventFillMap>(event, fValuesTrack);
+    VarManager::FillEvent<TEventFillMap>(event, fValuesDilepton);
+
+    // Set the global index offset to find the proper lepton
+    int indexOffset = -999;
+    for (auto dilepton : dileptons) {
+
+      int indexLepton1 = dilepton.index0Id();
+      int indexLepton2 = dilepton.index1Id();
+
+      if (indexOffset == -999) {
+        indexOffset = indexLepton1;
+      }
+
+      auto lepton1 = tracks.iteratorAt(indexLepton1 - indexOffset);
+      auto lepton2 = tracks.iteratorAt(indexLepton2 - indexOffset);
```

Comment:
I am not sure yet what the index offset is needed for, but you should add a note here to be removed later on after issues are solved.

Reply:
The index is needed becouse otherwise the interatorAt method does not recover the correct leptons from the track table. I would say that is just a temporary fix to solve the issue. I have added a further comment in the code